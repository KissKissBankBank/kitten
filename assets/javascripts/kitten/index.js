// Accessibility
export { VisuallyHidden } from './components/accessibility/visually-hidden'

// Accordeon
export { Accordeon } from './components/accordeon'

// Avatar
export { AvatarWithTextAndBadge } from './components/avatar/avatar-with-text-and-badge'

// Box
export { DocLinkBox } from './components/box/doc-link-box'
export { InformationBox } from './components/box/information-box'
export { LinkBox } from './components/box/link-box'

// Buttons
export { Button } from './components/buttons/button/button'
export { ButtonIcon } from './components/buttons/button-icon'
export { ButtonImageWithTextAndBadge } from './components/buttons/button-image-with-text-and-badge'
export { ButtonGroup } from './components/buttons/button-group'
export { ButtonImage } from './components/buttons/button-image'
export { ButtonQuestionMarkIcon } from './components/buttons/button-question-mark-icon'
export { CloseButton } from './components/buttons/close-button'
export {
  FacebookButtonIconWords,
  TwitterButtonIconWords,
  LinkedinButtonIconWords,
} from './components/buttons/social-button-icon-words'
export {
  FacebookButtonIcon,
  TwitterButtonIcon,
  LinkedinButtonIcon,
  InstagramButtonIcon,
  YoutubeButtonIcon,
  PinterestButtonIcon,
} from './components/buttons/social-button-icon'
export { TagButton } from './components/buttons/tag-button'
export { ButtonImageWithText } from './components/buttons/button-image-with-text'
export { CheckableButton } from './components/buttons/checkable-button'

// Cards
export { SimpleCard } from './components/cards/simple-card'
export { ArticleCard } from './components/cards/article-card'
export { BackerCard } from './components/cards/backer-card'
export { CartRewardCard } from './components/cards/cart-reward-card'
export { ContributionCard } from './components/cards/contribution-card'
export { DonationCard } from './components/cards/donation-card'
export { ProjectCard } from './components/cards/project-card'
export { ProjectCreatorCard } from './components/cards/project-creator-card'
export { ProjectSimilarCard } from './components/cards/project-similar-card'
export { RewardCard } from './components/cards/reward-card'
export { RewardEdition } from './components/cards/reward-edition'
export { SimilarProjectsCard } from './components/cards/similar-projects-card'
export { TeamCard } from './components/cards/team-card'
export { Triptych } from './components/cards/triptych'
export { VerticalCardWithAction } from './components/cards/vertical-card-with-action'
export { VerticalCard } from './components/cards/vertical-card'
export { ClickableCard } from './components/cards/clickable-card'
export { CrowdfundingCard } from './components/cards/crowdfunding-card'
export { EngagementCard } from './components/cards/engagement-card'

// Carousel
export { Carousel } from './components/carousel/carousel'
export { NavTabCarousel } from './components/carousel/nav-tab-carousel'
export { SimpleCarousel } from './components/carousel/simple-carousel'

// Comment
export { CommentAvatar } from './components/comments/comment-avatar'
export { CommentForm } from './components/comments/comment-form'
export { Comment } from './components/comments/comment'

// Date-picker
export { DatePicker } from './components/date-picker'
export { StandaloneRangeDatePicker } from './components/date-picker/standalone-range'

// Details
export { Details } from './components/details'

// Dev
export { DevGrid } from './components/dev/dev-grid'

// Dropdowns
export { DropdownButton } from './components/dropdowns/dropdown-button'
export { Dropdown } from './components/dropdowns/dropdown'
export { PhoneDropdown } from './components/dropdowns/phone-dropdown'
export { UserMenu } from './components/dropdowns/user-menu'

// Expandable
export { ExpandBoard } from './components/expandable/expand-board'

// Form
export { Field } from './components/form/field'
export { TextInputWithButton } from './components/form/text-input-with-button'
export { Checkbox } from './components/form/checkbox'
export { FormActions } from './components/form/form-actions'
export { FormAmountAndCurrency } from './components/form/form-amount-and-currency'
export { FormPhoneNumber } from './components/form/form-phone-number'
export { FormRow } from './components/form/form-row'
export { Label } from './components/form/label'
export { LocationInput } from './components/form/location-input'
export { PasswordInput } from './components/form/password-input'
export { RadioButtonSet } from './components/form/radio-button-set'
export { RadioButton } from './components/form/radio-button'
export { RichTextInput } from './components/form/rich-text-input'
export { Search } from './components/form/search'
export { SelectWithState } from './components/form/select-with-state'
export { Slider } from './components/form/slider'
export { Switch } from './components/form/switch'
export { TextInputAndSelect } from './components/form/text-input-and-select'
export { TextInputWithLimit } from './components/form/text-input-with-limit'
export { TextInputWithUnitForm } from './components/form/text-input-with-unit-form'
export { TextInputWithUnit } from './components/form/text-input-with-unit'
export { TextInput } from './components/form/text-input'

// Grid
export { Container } from './components/grid/container'
export { ScrollableContainer } from './components/grid/scrollable-container'
export { Grid, GridCol } from './components/grid/grid'
export { LegoGrid } from './components/grid/lego-grid'
export { Row } from './components/grid/row'
export { SideGrid } from './components/grid/side-grid'
export { StickyContainer } from './components/grid/sticky-container'
export { AdaptableGrid } from './components/grid/adaptable-grid'

// Headers
export { HeaderLogo } from './components/headers/header-logo'
export { HeaderTitles } from './components/headers/header-titles'
export { Header, HeaderItems, HeaderItem } from './components/headers/header'

// Heroes
export { Hero } from './components/heroes/hero'

// Images
export { ImageCropper } from './components/images/image-cropper'
export { ImageWithCaption } from './components/images/image-with-caption'

// Layout
export { ArrowContainer } from './components/layout/arrow-container'
export { IconContainer } from './components/layout/icon-container'
export { HorizontalStroke } from './components/layout/horizontal-stroke'
export { Line } from './components/layout/line'
export { Marger } from './components/layout/marger'
export { ResponsiveIframeContainer } from './components/layout/responsive-iframe-container'
export { Separator } from './components/layout/separator'
export { SideLayout } from './components/layout/side-layout'
export { VerticalStroke } from './components/layout/vertical-stroke'

// Lines
export { InfoLines } from './components/lines/info-lines'

// Links
export { ExternalRichLink } from './components/links/external-rich-link'
export { LinkList } from './components/links/link-list'

// Lists
export { IconList } from './components/lists/icon-list'
export { BulletList } from './components/lists/bullet-list'
export { List } from './components/lists/list'
export { CheckList } from './components/lists/check-list'
export { NavList } from './components/lists/nav-list'
export { TagList } from './components/lists/tag-list'
export { Timeline } from './components/lists/timeline'

// Loaders
export { LoaderWithParagraph } from './components/loaders/loader-with-paragraph'
export { Loader } from './components/loaders/loader'

// Menus
export { HeaderMenu } from './components/menus/header-menu'

// Meters
export { CounterBlock } from './components/meters/counter-block'
export { Progress } from './components/meters/progress'
export { ProgressRing } from './components/meters/progress-ring'

// Modals
export { Modal } from './components/modals/modal'
export { Overlay } from './components/modals/overlay'

// Navigation
export { HorizontalNav } from './components/navigation/horizontal-nav'
export { Pagination } from './components/navigation/pagination'
export { TabBar } from './components/navigation/tab-bar'
export { NavBar } from './components/navigation/nav-bar'

// Notifications
export { Alert } from './components/notifications/alert'
export { Badge } from './components/notifications/badge'
export { IconBadge } from './components/notifications/icon-badge'

// Pages
export { MarkdownPage } from './components/pages/markdown'

// Popovers
export { CallToActionPopover } from './components/popovers/call-to-action-popover'
export { Popover } from './components/popovers/popover'
export { SimplePopover } from './components/popovers/simple-popover'

// Search
export { SearchInput } from './components/search/search-input'

// Simulators
export { LoanSimulator } from './components/simulators/loan-simulator'

// Sliders
export { SliderBar } from './components/sliders/slider-bar'
export { SliderTooltip } from './components/sliders/slider-tooltip'
export { SliderWithTooltipAndPower } from './components/sliders/slider-with-tooltip-and-power'

// Status with Bullet
export { StatusWithBullet } from './components/status-with-bullet'

// Steppers
export { VerticalStepper } from './components/steppers/vertical-stepper'
export {
  StepperIconDefault,
  StepperIconInProgress,
  StepperIconValidated,
} from './components/steppers/stepper-icon'
export { Stepper } from './components/steppers/stepper'

// Tables
export { DoubleEntryTable } from './components/tables/double-entry-table'
export { ListTable } from './components/tables/list-table'
export { SingleEntryTable } from './components/tables/single-entry-table'

// Text Copy
export { TextCopy } from './components/text-copy'

// Tooltips
export { QuestionMarkWithTooltip } from './components/tooltips/question-mark-with-tooltip'
export { StaticTooltip } from './components/tooltips/static-tooltip'
export { TooltipNew } from './components/tooltips/tooltip-new'
export { Tooltip } from './components/tooltips/tooltip'

// Tours
export { HighlightHalo } from './components/tours/highlight-halo'
export { HeaderTour } from './components/tours/header-tour'
export { TourPopover } from './components/tours/tour-popover'
export { TourStep } from './components/tours/tour-step'
export { Tour } from './components/tours/tour'

// Typography
export { Paragraph } from './components/typography/paragraph'
export { SupTitle } from './components/typography/sup-title'
export { Text } from './components/typography/text'
export { Title } from './components/typography/title'
export { TitleWithStroke } from './components/typography/title-with-stroke'

// Uploaders
export { BasicUploader } from './components/uploaders/basic-uploader'
export { DocumentManager } from './components/uploaders/document-manager'
export { SimpleUploader } from './components/uploaders/simple-uploader'
export { Uploader } from './components/uploaders/uploader'

// Videos
export { Video } from './components/videos/video'
export { GifVideo } from './components/videos/gif-video'
export { EmbedPlayer } from './components/videos/embed-player'
