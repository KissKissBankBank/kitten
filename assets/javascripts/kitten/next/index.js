export { Modal } from '../components/modals/next'
export { Stepper } from '../components/steppers/next/stepper'
export { Button } from '../components/buttons/button/button'
export { Carousel } from '../components/carousel/carousel/carousel'
export { Paragraph } from '../components/typography/paragraph/next'
