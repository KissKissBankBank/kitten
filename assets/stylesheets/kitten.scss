@import 'kitten/helpers';
@import 'kitten/schema';
