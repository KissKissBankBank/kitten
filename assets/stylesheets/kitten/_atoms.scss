@import 'kitten/atoms/buttons';
@import 'kitten/atoms/form';
@import 'kitten/atoms/typography';
