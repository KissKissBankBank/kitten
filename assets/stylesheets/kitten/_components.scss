// Buttons
@import 'kitten/components/buttons/button-icon';
@import 'kitten/components/buttons/button-image';
@import 'kitten/components/buttons/button-image-with-text';
@import 'kitten/components/buttons/button-with-badge';
@import 'kitten/components/buttons/button';
@import 'kitten/components/buttons/tag-button';

// Box
@import 'kitten/components/box/link-box';
@import 'kitten/components/box/information-box';

// Cards
@import 'kitten/components/cards/card';

// Dev
@import 'kitten/components/dev/dev-breakpoint';
@import 'kitten/components/dev/dev-grid';

// Dropdowns
@import 'kitten/components/dropdowns/dropdown';
@import 'kitten/components/dropdowns/phone-dropdown';

// Footer
@import 'kitten/components/footer/footer';

// Form
@import 'kitten/components/form/checkbox';
@import 'kitten/components/form/location-input';
@import 'kitten/components/form/form-actions';
@import 'kitten/components/form/form-composer';
@import 'kitten/components/form/form-row';
@import 'kitten/components/form/label';
@import 'kitten/components/form/radio-button';
@import 'kitten/components/form/search-input';
@import 'kitten/components/form/select';
@import 'kitten/components/form/slider-tooltip';
@import 'kitten/components/form/slider';
@import 'kitten/components/form/switch';
@import 'kitten/components/form/text-area-wrapper';
@import 'kitten/components/form/text-input';
@import 'kitten/components/form/text-input-with-button';
@import 'kitten/components/form/text-input-with-limit';
@import 'kitten/components/form/text-input-with-info';
@import 'kitten/components/form/text-input-with-unit';

// Grid
@import 'kitten/components/grid/grid';
@import "kitten/components/grid/row";
@import "kitten/components/grid/side-grid";
@import "kitten/components/grid/single-column";
@import "kitten/components/grid/vertical-grid";

// Headers
@import 'kitten/components/headers/header';

// Heroes
@import "kitten/components/heroes/hero";

// Layout
@import 'kitten/components/layout/line';

// Links
@import 'kitten/components/links/external-rich-link';
@import 'kitten/components/links/link-list';

// Lists
@import 'kitten/components/lists/bullet-list';

// Meters
@import 'kitten/components/meters/progress';

// Navigation
@import 'kitten/components/navigation/horizontal-nav';
@import 'kitten/components/navigation/platform-switch';
@import 'kitten/components/navigation/tab-bar';
@import 'kitten/components/navigation/user-menu';

// Notifications
@import 'kitten/components/notifications/alert';
@import 'kitten/components/notifications/badge';

// Press
@import 'kitten/components/press/press-quotes';

// Popovers
@import 'kitten/components/popovers/popover';

// Simulators
@import 'kitten/components/simulators/loan-simulator';
@import 'kitten/components/simulators/simulator-result';

// Steppers
@import 'kitten/components/steppers/stepper';

// Teaser
@import "kitten/components/teaser/teaser";

// Testimonies
@import "kitten/components/testimonies/testimony-list";
@import "kitten/components/testimonies/testimony";

// Titles
@import 'kitten/components/titles/title-with-actions';

// Tooltips
@import 'kitten/components/tooltips/tooltip';

// Tours
@import 'kitten/components/tours/tour';
@import 'kitten/components/tours/header-tour';

// Typography
@import 'kitten/components/typography/body-copy';
@import 'kitten/components/typography/paragraph';
@import 'kitten/components/typography/title';
