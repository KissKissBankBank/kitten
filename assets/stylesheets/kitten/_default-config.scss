// Media queries

// Media queries sorted by thinnest to largest
$k-media-queries: (
  xxs: (
    min-width: null,
    max-width: 479px,
  ),
  xs: (
    min-width: 480px,
    max-width: 639px,
  ),
  s: (
    min-width: 640px,
    max-width: 767px,
  ),
  m: (
    min-width: 768px,
    max-width: 1079px,
  ),
  l: (
    min-width: 1080px,
    max-width: 1439px,
  ),
  xl: (
    min-width: 1440px,
    max-width: null,
  ),
) !default;

// Grid

$k-grid: (
  columns: 12,
  container: 1440px,
  gutter-width: 20px,
) !default;

// Susy

$columns: map-get($k-grid, 'columns');
$container: map-get($k-grid, 'container');
$gutter-width: map-get($k-grid, 'gutter-width');
$column-width: ($container - ($columns - 1) * $gutter-width) / $columns;

$susy: (
  columns: $columns,
  container: $container,
  gutter-position: split,
  gutters: $gutter-width/$column-width,
) !default;

// Fonts

$k-fonts: (
  source-sans-light: (
    family: ("Source Sans Pro", Helvetica, Arial, sans-serif),
    weight: 300
  ),
  source-sans: (
    family: ("Source Sans Pro", Helvetica, Arial, sans-serif),
    weight: 400
  ),
  source-sans-semi-bold: (
    family: ("Source Sans Pro", Helvetica, Arial, sans-serif),
    weight: 600
  ),
  source-sans-uppercase: (
    family: ("Source Sans Pro", Helvetica, Arial, sans-serif),
    weight: 400,
    transform: uppercase
  ),
  source-sans-semi-bold-uppercase: (
    family: ("Source Sans Pro", Helvetica, Arial, sans-serif),
    weight: 600,
    letter-spacing: .1rem,
    transform: uppercase
  ),
  dosis: (
    family: ("Dosis", Helvetica, Arial, sans-serif),
    weight: 400
  )
) !default;

// Colors

$k-colors: (
  'font-1': #333,
  'font-2': #b8b8b8,
  'background-1': #fff,
  'background-2': #fbfbfb,
  'background-3': #f6f6f6,
  'line-1': #eee,
  'line-2': #d8d8d8,

  'primary-1': #4a84ff,
  'primary-2': #3b75f0,
  'primary-3': #2c66e1,
  'primary-4': #cadbfd,
  'primary-5': #ecf2ff,

  'secondary-1': #4bfffa,
  'secondary-2': #3cf0eb,
  'secondary-3': #2de1dc,

  'tertiary-1': #68ffa0,
  'tertiary-2': #59f091,
  'tertiary-3': #4ae191,

  'valid': #70c050,
  'error': #ff4146,
) !default;
