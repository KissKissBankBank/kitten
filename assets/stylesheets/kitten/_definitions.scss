@import 'kitten/definitions/common/color';
@import 'kitten/definitions/common/number';
@import 'kitten/definitions/common/string';
@import 'kitten/definitions/font';
@import 'kitten/definitions/button-colors';
