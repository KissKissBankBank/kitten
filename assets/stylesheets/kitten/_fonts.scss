/* Maax-Regular */
@font-face {
  font-family: 'Maax';
  font-weight: 400;
  font-style: normal;

  src: url('~maax/Maax-Regular.eot') format('eot'),
       url('~maax/Maax-Regular.svg') format('svg'),
       url('~maax/Maax-Regular.ttf') format('ttf'),
       url('~maax/Maax-Regular.woff2') format('woff2'),
       url('~maax/Maax-Regular.woff') format('woff');
}

/* Maax-Medium */
@font-face {
  font-family: 'Maax';
  font-weight: 500;
  font-style: normal;

  src: url('~maax/Maax-Medium.eot') format('eot'),
       url('~maax/Maax-Medium.svg') format('svg'),
       url('~maax/Maax-Medium.ttf') format('ttf'),
       url('~maax/Maax-Medium.woff2') format('woff2'),
       url('~maax/Maax-Medium.woff') format('woff');
}

/* Maax-Bold */
@font-face {
  font-family: 'Maax';
  font-weight: 600;
  font-style: normal;

  src: url('~maax/Maax-Bold.eot') format('eot'),
       url('~maax/Maax-Bold.svg') format('svg'),
       url('~maax/Maax-Bold.ttf') format('ttf'),
       url('~maax/Maax-Bold.woff2') format('woff2'),
       url('~maax/Maax-Bold.woff') format('woff');
}
