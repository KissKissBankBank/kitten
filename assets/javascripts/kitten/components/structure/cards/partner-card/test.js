import React from 'react'
import renderer from 'react-test-renderer'
import { PartnerCard } from './index'
import { Text } from 'kitten'

describe('<PartnerCard />', () => {
  let component

  describe('by default', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <PartnerCard
            status="normal"
            imageProps={{
              src: 'kitten.jpg',
              alt: 'image alt',
            }}
          >
            <PartnerCard.Avatar
              imageProps={{
                src: 'kitten.jpg',
                alt: 'image alt',
              }}
            />
            <PartnerCard.Line>
              <Text size="small" weight="500">
                Les Jeunes Pousses méritent tout notre soutien : Naturalia s’engage dans le développement de votre produit bio.
              </Text>
              <PartnerCard.Sticker>
                Citoyenneté
              </PartnerCard.Sticker>
              <Text
                tag="p"
                weight="700"
                size="small"
                className="k-u-link-font1 k-u-cursor-pointer k-u-margin-none"
              >
                En savoir plus
              </Text>
            </PartnerCard.Line>
          </PartnerCard>,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
