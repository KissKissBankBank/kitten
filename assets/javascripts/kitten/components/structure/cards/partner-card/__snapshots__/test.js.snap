// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PartnerCard /> by default matches with snapshot 1`] = `
<div
  className="sc-gsnTZi hVAcRK"
>
  <div
    className="k-PartnerCard__image"
  >
    <img
      alt="image alt"
      className="k-PartnerCard__image__image"
      src="kitten.jpg"
    />
  </div>
  <div
    className="k-PartnerCard__content"
  >
    <div
      className="k-PartnerCard__avatar"
    >
      <img
        alt="image alt"
        src="kitten.jpg"
      />
    </div>
    <div
      className="k-PartnerCard__line"
    >
      <span
        className="k-u-size-small k-u-weight-500"
        style={Object {}}
      >
        Les Jeunes Pousses méritent tout notre soutien : Naturalia s’engage dans le développement de votre produit bio.
      </span>
      <div
        className="k-PartnerCard__sticker"
      >
        <span
          className="sc-bczRLJ bKqmxm k-Tag k-u-margin-bottom-double k-u-margin-top-double k-u-ellipsis k-Tag--disabled k-Tag--medium k-Tag--status"
        >
          Citoyenneté
        </span>
      </div>
      <p
        className="k-u-size-small k-u-weight-700 k-u-link-font1 k-u-cursor-pointer k-u-margin-none"
        style={Object {}}
      >
        En savoir plus
      </p>
    </div>
  </div>
</div>
`;
